### ISSUE TITLE
qt client accessibility bugs

### ISSUE THREAD

**Issue by amirmahdifard0** [BearWare/TeamTalk5#957]
Hello. I tried qt client teamtalk for the first time today and want to report its accessibility issues. One of the problems is that when we are on the message history and press the tab key once to reach the channels list, the screen reeder does not read the channels list for us unless we press one of the directions that the channels for We are displayed. The same is true for files list. Another problem is that when I'm in the program, nvda constantly reads percentages for me, unfortunately as I searched I did not find the source and realized that this is a bug. And another small problem is that it reads some buttons twice. For example dounload butten dounload. That was it now. I hibernate until I find new shapes. Bye!

Windows platform

5.8

qt

**Comment by Oreonan** [comment]
1. For the issue with tree names, it's unfortunately a QT issue, we are aware of this but we can't fix this. I reported the bug to QT and it'll be fix in a future QT release.
2. For the percentages, it's the percentages of your microphone level, please go to Preferences/Display and uncheck the option "Enable VU-meter updates"
3. For button announce twice, I don't think it's really a bug, I try to explain why: basically, this buttons doesn't have text, just an icon, and a tooltip when mouse move on one of this. For accessibility, I added a text, only visible by screenreaders, to this buttons. Now some screenreaders like NVDA read text accessible, ad also tooltip, we can't really change this because if we remove text accessible, some screenreaders will don't read this buttons anymore.
I hope my answers can help you, please confirm if the second point is resolve after uncheck option.
Thanks for your reports, we try to make QT client as accessible as the classic client and tests like this are really helpful.

**Comment by amirmahdifard0** [comment]
Thank you for your reply. My second problem was solved with what you said. But please note that if you make the qt client available as classic as possible, do not delete the classic version at all. Because these two clients have a completely different user interface, and maybe one person with a classic user interface is more comfortable and another person with a qt user interface.

**Comment by amirmahdifard0** [comment]
But a question has occupied my mind. How did you make the classic client so well accessible from the start when the qt client accessibility could not be so well classic

**Comment by Oreonan** [comment]
Good if your second problem is now resolved.
For now, classic client is in a kind of extended support, unless a feature is trivial for us to add to classic, classic client will only receive bugs fixes. Previously we had to implement all features for twice that take some time.
For your latest question classic is based on a development framework who use basic Windows components, so it's probably as accessible as Windows system :)
Maybe Bjørn will be able to give you more informations if you want. I started recently to contribute and I don't know all the history of all clients :)

**Comment by amirmahdifard0** [comment]
I did not understand what you mean, you do not add new features to the classic?

**Comment by bear101** [comment]
Windows has a set of [base components](https://docs.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-createwindowa) for buttons, groupbox, combobox, etc. These work very well with accessibility because they have been in Windows right since the start and screen-reader have adapted to these base components.

In Qt framework they have made their own abstraction of these base components in order to be portable to other platforms (Linux, macOS, Android, etc). This means that in order for the screen-readers to know if it's a button then the screen-reader must support Qt's implementation of these base components. Alternatively the Qt framework has to inform the screen reader what is currently active in the application.

So basically it's a struggle between screen-readers adapting to Qt or Qt adapting to screen-readers.
